### DISCUSSION TITLE
possible future api?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#30236] [Copilot Conversations]
i don't use any of the platforms copilot currently supports (vs code, vim, etc). in the future could copilot have an api (or get merged into the github api), so it's decoupled from any particular ide?

say behavior like:

```
from github import copilot

payload = copilot("""
# generate some code
def sum_nums_1_to_n(n):
  pass
"""
)

**Comment by zbirenbaum** [comment]
Internally copilot is just a fine-tuned openai model and uses the same api as it. I wouldn't hold my breath on copilot having something like this, but you can definitely make your own api calls if you get an OpenAI codex key.

**Comment by zorzigio** [comment]
@zbirenbaum for people already paying for GitHub Copilot, would it not make sense to be able to create their own tools leveraging Copilot?

For instance, would be great to have a bash completion running on Copilot directly integrated in the terminal.
This already exists with Codex, but why do I have to sign up to Codex if I already have Copilot?

**Comment by zbirenbaum** [comment]
> @zbirenbaum for people already paying for GitHub Copilot, would it not make sense to be able to create their own tools leveraging Copilot?

I agree, which is why I wrote my own tools. Check out copilot.lua. 

> This already exists with Codex, but why do I have to sign up to Codex if I already have Copilot?

I ended up writing a LSP for codex but it wasn't feasible to use like copilot due to rate limiting. The whole issue is that Codex had documentation and is designed to be built on while copilot does not, so everything has to be reverse engineered.

**Comment by Mini256** [comment]
Hi, @davecheney, I want to consult you some questions about Copilot API. 

Our team has an idea. We want to implement a chat robot (similar to Hey GitHub) that executes SQL statements in the database through dialogue on our website. We want to call the internal API of the copilot to implement the text to SQL feature, but I'm not sure whether this can be allowed by GitHub?

**Reply by davecheney** [reply]
@Mini256 i'm sorry, we have no plans for a public API at this time.

**Reply by Mini256** [reply]
😢 So sad. Thanks for your respond.
